scripts/build-renode.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+RENODE_BIN="renode"
+
+command -v $RENODE_BIN 2>&1 1>/dev/null
+if [ $? -ne 0 ]; then
+	echo "$RENODE_BIN not found. Please install it and make sure it's available in PATH before running this script."
+	exit 1
+fi
+
+set -x
+set -e
+
+case $CPU in
+	vexriscv)
+		RENODE_PLATFORM=litex_vexriscv.repl
+		;;
+	picorv32)
+		RENODE_PLATFORM=litex_picorv32.repl
+		;;
+	*)
+		echo "CPU $CPU isn't supported at the moment."
+		exit 1
+		;;
+esac
+
+if [ ! -d $TARGET_BUILD_DIR/software/include/generated ]; then
+	make firmware
+fi
+
+script_file=$(mktemp /tmp/litex_buildenv.resc.XXXXXX)
+function cleanup {
+	rm -rf "$script_file"
+}
+trap cleanup EXIT
+
+function add_to_script {
+	echo $1 >> "$script_file"
+}
+
+add_to_script "using sysbus"
+add_to_script "mach create \"litex-$CPU\""
+add_to_script "machine LoadPlatformDescription @platforms/cpus/$RENODE_PLATFORM"
+add_to_script "cpu StartGdbServer 10001"
+add_to_script "showAnalyzer sysbus.uart"
+
+# BIOS
+if grep -q 'ROM_BASE 0x00000000' $TARGET_BUILD_DIR/software/include/generated/mem.h; then
+	echo "Platform has BIOS ROM, adding BIOS"
+
+	add_to_script "sysbus LoadBinary @$TARGET_BUILD_DIR/software/bios/bios.bin 0x0"
+	add_to_script "cpu PC 0x0"
```

Comment:
The reset vector isn't always zero.

Reply:
https://github.com/timvideos/HDMI2USB-firmware-prebuilt/blob/52959d0bfe90b7536da7e516120c586e6ea3aafc/archive/master/v0.0.4-455-gde59df1/ice40_up5k_b_evn/base/lm32.minimal/software/include/generated/csr.h#L322-L325

Reply:
In the fixed version, I assumed the reset vector value to be the same as `ROM_BASE`.